adding puppi to miniAODs

jshlee opened the pull request:
added puppi jet collection and puppi met into miniAODs

added jet charge, btagging, secondary vertex info and daughter links for puppi jets

added puppi weights as packed 8 bit into packcandidates

removed labels in puppi producer since only three different types of objects are produced

fixed the packing for liblogintpack, negative values were being lost

updated PackedCandidate ClassVersion to 17

added new iputs to PATPackedCandidateProducer, (PuppiWeight, PuppiCandsMap and PuppiCands) to save puppi weights into packcandidates and ValueMap for puppi to be able to map back to the puppi candidates from puppi jets to packcandidates

jshlee committed:
adding puppi to miniAODs
added puppi jets and met
added puppi weights to packcandidates

cmsbuild commented:
A new Pull Request was created by @jshlee (Jason Lee) for CMSSW_7_5_X.

adding puppi to miniAODs

It involves the following packages:

DataFormats/PatCandidates
PhysicsTools/PatAlgos


@cmsbuild, @vadler, @nclopezo, @monttj can you please review it and eventually sign? Thanks.
@rappoccio, @imarches, @ahinzmann, @acaudron, @mmarionncern, @jdolen, @nhanvtran, @schoef, @ferencek, @mariadalfonso, @pvmulder, @TaiSakuma this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@nclopezo, @ktf you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

jshlee committed:
removed bool for checking if collection exists

cmsbuild commented:
Pull request #8122 was updated. @cmsbuild, @vadler, @nclopezo, @monttj can you please check and sign again.

arizzi commented:
I'm testing the size effect 

adding @gpetruc

arizzi commented on PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py:
```diff
@@ -132,7 +132,43 @@ def miniAOD_customizeCommon(process):
+
+    addJetCollection(process, postfix   = "", labelName = 'Puppi', jetSource = cms.InputTag('ak4PFJetsPuppi'),
+                    jetCorrections = ('AK4PF', ['L1FastJet', 'L2Relative', 'L3Absolute'], ''),
+                    algo= 'AK', rParam = 0.4)
```
b-tagging should be enabled here

arizzi commented:
added b-tagging and merged with #8132 on https://github.com/arizzi/cmssw/tree/miniAOD-pvassociation-packedsorting-puppi

now testing overall size

jshlee committed:
added jet charge, btagging, secondary vertex info for puppi jets

cmsbuild commented:
Pull request #8122 was updated. @cmsbuild, @vadler, @nclopezo, @monttj can you please check and sign again.

arizzi commented on DataFormats/PatCandidates/src/PackedCandidate.cc:
```diff
@@ -268,6 +268,8 @@ bool pat::PackedCandidate::longLived() const {return false;}
 bool pat::PackedCandidate::massConstraint() const {return false;}
 
+// puppiweight
+void pat::PackedCandidate::setPuppiWeight(float p) { packedPuppiweight_ = pack8log(p,-2,0);}
```
how about making it symmetric in -1,1 instead? so that it has higher precision near zero too?
pack8log( (p-0.5)*2,-2,0,64)
where the last argument (64) is needed if you want to take the same space as it is doing now with 7+1 signed bits (i.e. one is never used)

arizzi commented:
concerning the daughters of jets ideally one should create the value map
PuppiCandidates -> original PF Candidates here:
CommonTools/PileupAlgos/plugins/PuppiProducer.cc
and then read it in PATPackedCandidateProducer and add to the already produced value map also the 
puppi -> packed (this value map already contains original PF -> packed and trackRefs to packed)

jshlee committed:
changed packed puppi weight
to pack8log( (p-0.5)*2,-2,0,64)
liblogintpack bug fix - negative not saved for r=0 or r=base-1

jshlee committed:
changed btagDiscriminators mapped to
process.patJets.discriminatorSources

arizzi commented:
here some size estimate including also #8132 (and fixing the conflicts)
http://arizzi.web.cern.ch/arizzi/puppipuppi.html

we still miss the daughter vector to be properly rekeyed (it should take another ~200-300 bytes/event)

cmsbuild commented:
Pull request #8122 was updated. @cmsbuild, @vadler, @nclopezo, @monttj can you please check and sign again.

cmsbuild commented:
-1
@jshlee This pull request cannot be automatically  merged, could you please rebase it?

cmsbuild commented:
Pull request #8122 was updated. @diguida, @lveldere, @ianna, @cerminar, @covarell, @bendavid, @civanch, @franzoni, @cmsbuild, @ktf, @davidlange6, @vciulli, @Dr15Jones, @cvuosalo, @mdhildreth, @deguio, @slava77, @ggovi, @vadler, @mmusich, @mulhearn, @ssekmen, @nclopezo, @danduggan, @boudoul, @thuer, @monttj, @srimanob can you please check and sign again.

lveldere commented:
can you rebase?
thanks

jshlee commented:
@lveldere I tried, but I must have rebased wrong.
I'll try again

cmsbuild commented:
Pull request #8122 was updated. @diguida, @lveldere, @ianna, @cerminar, @covarell, @bendavid, @civanch, @franzoni, @cmsbuild, @ktf, @davidlange6, @vciulli, @Dr15Jones, @cvuosalo, @mdhildreth, @deguio, @slava77, @ggovi, @vadler, @mmusich, @mulhearn, @ssekmen, @nclopezo, @danduggan, @boudoul, @thuer, @monttj, @srimanob can you please check and sign again.

arizzi commented:
@jshlee how about the daughters? are you  trying to get them re-keyed?

jshlee commented:
@arizzi, yep I'm still working on this.

On 9 Mar 2015, at 6:31 pm, arizzi <notifications@github.com<mailto:notifications@github.com>> wrote:


@jshlee<https://github.com/jshlee> how about the daughters? are you trying to get them re-keyed?

—
Reply to this email directly or view it on GitHub<https://github.com/cms-sw/cmssw/pull/8122#issuecomment-77822793>.

davidlange6 commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 03410adb881cdc5a6566f79a5d63a5d645cbda93
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8122/3225/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8122/3225/summary.html

jshlee committed:
trying to add puppiJet Daughters

cmsbuild commented:
+1
Tested at: 03410adb881cdc5a6566f79a5d63a5d645cbda93
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8122/3258/summary.html

jshlee committed:
Update PuppiProducer.cc

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8122/3258/summary.html

arizzi committed:
fixing the valumaps

arizzi commented:
@jshlee the branch seem a bit a rebase nightmare... there are the same commits a couple of times ... can you please clean it up?

cmsbuild commented:
Pull request #8122 was updated. @nclopezo, @cvuosalo, @monttj, @cmsbuild, @slava77, @vadler can you please check and sign again.

cmsbuild commented:
+1
Tested at: 60cc13a1fff9693488325f5023b9a7c9d4394b06
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8122/2206/summary.html

arizzi commented:
this are the expected conflicts with just merged #8132 
@jshlee can you fix as we discussed (and open a 74X PR too)

jshlee commented:
@arizzi yep will do

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8122/2206/summary.html

jshlee committed:
adding PuppiSrc and PuppiSrcMap to packedPFCandidates_cfi

cmsbuild commented:
Pull request #8122 was updated. @nclopezo, @cvuosalo, @monttj, @cmsbuild, @slava77, @vadler can you please check and sign again.

cmsbuild commented:
Pull request #8122 was updated. @nclopezo, @cvuosalo, @monttj, @cmsbuild, @slava77, @vadler can you please check and sign again.

jshlee commented:
rebased to fix conflict with #8132

arizzi commented on PhysicsTools/PatAlgos/plugins/PATPackedCandidateProducer.cc:
```diff
@@ -104,6 +111,39 @@ void pat::PATPackedCandidateProducer::produce(edm::Event& iEvent, const edm::Eve
+    iEvent.getByToken( PuppiCands_, puppiCands );
+    std::vector<int> mappingPuppi(puppiCands->size());
+
+    std::vector<pat::PackedCandidate::PVAssoc> fromPV(cands->size(), pat::PackedCandidate::NoPV);
```
this part is obsolete and was removed by #8132, it should not be added back

arizzi commented on PhysicsTools/PatAlgos/python/slimming/packedPFCandidates_cfi.py:
```diff
@@ -6,5 +6,8 @@
     vertexAssociator = cms.InputTag("primaryVertexAssociation","original"),
+    PuppiWeight = cms.InputTag("puppi","PuppiWeights"),
+    PuppiSrc = cms.InputTag("puppi"),
+    PuppiSrcMap = cms.InputTag("puppi"),
```
we do not need two input tags here, one is enough as the producer is expected to be the same (in fact, is the label "PuppiWeights" really needed ???)

jshlee replied:
PuppiWeights isnt needed since puppi only makes one float value map.
for the input tags for PuppiSrc and PuppiSrcMap, since the edmTokens where different, I made both. I guess I can change them to just one puppiSrc

arizzi replied:
@nhanvtran do you mind if we remove the label? or is there a reason for it?

nhanvtran replied:
It's fine for me to remove it

arizzi commented on PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py:
```diff
@@ -174,6 +174,58 @@ def miniAOD_customizeCommon(process):
 
+    # Adding puppi jets
+    process.load('CommonTools.PileupAlgos.Puppi_cff')
+    process.patCandidates += process.puppi
```
we are running unscheduled... what is this for?
btw I do not see a keep statement for slimmedJetsPuppi, how are jets store?

jshlee replied:
I was making sure that puppi was run before the packcandidates were made. It should work without it.
slimmedJetsPuppi and slimmedMETsPuppi are saved due to 
        'keep *_slimmedJets*_*_*',
        'keep *_slimmedMETs*_*_*',
in slimming_cff

arizzi replied:
3be2279f from @rappoccio in the subjet PR is actually removing such star for jets, can you please add a specific line?

jshlee committed:
removing obsolete part that was removed in #8132
but added back during merging

cmsbuild commented:
Pull request #8122 was updated. @nclopezo, @cvuosalo, @monttj, @cmsbuild, @slava77, @vadler can you please check and sign again.

jshlee committed:
removing the label in puppi producer

cmsbuild commented:
Pull request #8122 was updated. @nclopezo, @cvuosalo, @monttj, @cmsbuild, @slava77, @vadler can you please check and sign again.

slava77 commented:
@cmsbuild please test

Andrea, are you happy with this PR now?

cmsbuild commented:
The tests are being triggered in jenkins.

arizzi commented:
Yep seem final now (I did not check after the last commit bit from the
diffs looks OK)
Il 11/mar/2015 19:18 "cmsbuild" <notifications@github.com> ha scritto:

> The tests are being triggered in jenkins.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/8122#issuecomment-78336105>.
>

cmsbuild commented:
-1 
Tested at: 70be3b4c462354f764ecaa8fdbf8885cfc1a022c
 I found an error when building: 
 
 <pre>                 from /afs/cern.ch/cms/sw/ReleaseCandidates/vol1/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-03-11-0200/src/DataFormats/Provenance/interface/Provenance.h:11,
                 from /afs/cern.ch/cms/sw/ReleaseCandidates/vol1/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-03-11-0200/src/DataFormats/Common/interface/BasicHandle.h:29,
                 from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-11-0200/src/FWCore/Framework/interface/Event.h:20,
                 from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-11-0200/src/TopQuarkAnalysis/Examples/plugins/TopJetAnalyzer.h:6,
                 from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-11-0200/src/TopQuarkAnalysis/Examples/plugins/TopJetAnalyzer.cc:1:
/afs/cern.ch/cms/sw/ReleaseCandidates/vol1/slc6_amd64_gcc491/lcg/root/5.34.22-cms3/include/Reflex/Member.h:552:40: fatal error: Reflex/internal/MemberBase.h: No such file or directory
compilation terminated.
In file included from /afs/cern.ch/cms/sw/ReleaseCandidates/vol1/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-03-11-0200/src/DataFormats/Candidate/interface/Candidate.h:17:0,
                 from /afs/cern.ch/cms/sw/ReleaseCandidates/vol1/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-03-11-0200/src/CommonTools/CandUtils/interface/pdgIdUtils.h:3,
                 from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-11-0200/src/AnalysisDataFormats/TopObjects/interface/TtGenEvent.h:4,
                 from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-11-0200/src/TopQuarkAnalysis/Examples/plugins/TopGenEventAnalyzer.h:13,
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8122/2748/summary.html

ktf commented:
Please do not add a direct dependency on reflex to fix this build error, 
but rather one on one of the packages which still depend on it (e.g. 
DataFormats/Math).

cmsbuild commented:
+1
Tested at: 37264643b91477566f0f9e9c3853bbfc9ba43029
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8122/6554/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8122/6554/summary.html

arizzi commented:
Ciao Giulio, I'm not understanding what the error actually is and how can
it be related to this PR
Il 12/mar/2015 02:22 "cmsbuild" <notifications@github.com> ha scritto:

> Comparison is ready
>
> https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8122/6554/summary.html
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/8122#issuecomment-78407387>.
>

jshlee commented:
@ktf is any further action required??

arizzi commented:
@monttj @slava77  can you guys sign this or comment if you have any issue with it?

slava77 commented:
it looks like the branch was just or updated otherwise

jshlee commented:
I rebased so that it is in sync with 74X PR
can you test again and sign this

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #8122 was updated. @nclopezo, @cvuosalo, @monttj, @cmsbuild, @slava77, @vadler can you please check and sign again.

arizzi commented:
I think when @slava77 said in sync  just meant "same commits as there" not really that they have to share the branch or anything like that. Is there a new 74X PR? (btw when you open PR please reference the relevant PRs there, eg. this, and add relevant people, e.g. me and giovanni, as well as describe what the PR is for future reference)

jshlee commented:
PR for 74X is in #8249
ahh ok.... will do so in future.
cheers,
jason.

gpetruc committed:
Merged refs/pull/8249/head from repository cms-sw

gpetruc committed:
Better packing of puppi weights, preserving zero and one

gpetruc committed:
Fix the linking of puppi candidates after sorting (and also of tracks, since it had been broken by mistake in #8133)

jshlee committed:
Merge pull request #296 from gpetruc/fixPuppi_74X

Fix daughter references, and better packing of puppi weights.

cmsbuild commented:
Pull request #8122 was updated. @nclopezo, @cvuosalo, @monttj, @cmsbuild, @slava77, @vadler can you please check and sign again.

arizzi committed:
do not ignore the input puppi value map

jshlee committed:
Merge pull request #297 from arizzi/puppi

do not ignore the input puppi value map

cmsbuild commented:
Pull request #8122 was updated. @nclopezo, @cvuosalo, @monttj, @cmsbuild, @slava77, @vadler can you please check and sign again.

jshlee commented:
andrea was too quick for me

monttj commented:
+1

arizzi commented:
@jshlee can you cross check that the re-keying works properly? i.e. the daughters list is correct reading back the puppi jets

arizzi commented:
it may still be broken (Giovanni's version was ok but mine seem not)... we need to sort this out... I can check over the weekend or monday

jshlee commented:
I've check both and they are fine.

arizzi commented:
ok works for me too on a clean setup.
at this point I think this PR is complete and should just be signed and merged.

cmsbuild commented:
+1
Tested at: ad33789565d002dd3328a2a67c065cd4c894f831
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8122/4721/summary.html

cvuosalo commented:
Starting matrix tests since Jenkins results are incomplete.

cmsbuild commented:
-1
@jshlee This pull request cannot be automatically  merged, could you please rebase it?

cvuosalo commented:
+1

For #8122 ad33789565d002dd3328a2a67c065cd4c894f831

This PR adds PUPPI (pile-up per particle ID) weights to MiniAOD. The simple addition of new content to MiniAOD should not change the calculation of reco quantities. This PR is the same as #8249, which is aimed at 74X, and which was also approved by Reco.

The Reco code change is satisfactory. Jenkins tests are OK, except that the comparison against the baseline did not complete. Short matrix tests against baseline CMSSW_7_4_X_2015-03-13-1400 show no significant differences.

It appears this PR is failing the merge check by cmsbuild. Provided that problem can be surmounted, Reco approves this PR.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_5_X IBs unless changes (but tests are reportedly failing). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @nclopezo, @ktf, @smuzaffar

arizzi commented:
@cmsbuild github seem to say that is mergable... is really another rebase needed??

davidlange6 commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_5_X IBs unless changes or unless it breaks tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @nclopezo, @ktf, @smuzaffar

davidlange6 commented on DataFormats/PatCandidates/src/classes_def_objects.xml:
```diff
@@ -252,8 +252,8 @@
-  <class name="pat::PackedCandidate" ClassVersion="16">
-   <version ClassVersion="17" checksum="1633877368"/>
+  <class name="pat::PackedCandidate" ClassVersion="17">
+    <version ClassVersion="17" checksum="1257500115"/>
```
looks like this should be 18 instead of 17 (interfering PRs again...)

arizzi replied:
well, does it matter? I mean if no release exists with "17" why should we care about increasing the version number?

davidlange6 replied:
your question assumes a particular future path of the release builds. While likely true, lets try not to repeat mistakes of the past…

> On Mar 15, 2015, at 6:39 PM, arizzi <notifications@github.com> wrote:
> 
> In DataFormats/PatCandidates/src/classes_def_objects.xml:
> 
> > @@ -252,8 +252,8 @@
> >     <version ClassVersion="11" checksum="1239840459"/>
> >    </class>
> >  
> > -  <class name="pat::PackedCandidate" ClassVersion="16">
> > -   <version ClassVersion="17" checksum="1633877368"/>
> > +  <class name="pat::PackedCandidate" ClassVersion="17">
> > +    <version ClassVersion="17" checksum="1257500115"/>
> 
> well, does it matter? I mean if no release exists with "17" why should we care about increasing the version number?
> 
> —
> Reply to this email directly or view it on GitHub.
>

arizzi replied:
mhh in any case I expect the checksum to be neither of the two if both PRs
are changing something... so is not just a matter of increasing 17 to 18,
right?

On Sun, Mar 15, 2015 at 6:49 PM, David Lange <notifications@github.com>
wrote:

> In DataFormats/PatCandidates/src/classes_def_objects.xml
> <https://github.com/cms-sw/cmssw/pull/8122#discussion_r26450034>:
>
> > @@ -252,8 +252,8 @@
> >     <version ClassVersion="11" checksum="1239840459"/>
> >    </class>
> >
> > -  <class name="pat::PackedCandidate" ClassVersion="16">
> > -   <version ClassVersion="17" checksum="1633877368"/>
> > +  <class name="pat::PackedCandidate" ClassVersion="17">
> > +    <version ClassVersion="17" checksum="1257500115"/>
>
> your question assumes a particular future path of the release builds.
> While likely true, lets try not to repeat mistakes of the past…
> … <#14c1e8e21ca76eb6_>
>  On Mar 15, 2015, at 6:39 PM, arizzi <notifications@github.com> wrote: In
> DataFormats/PatCandidates/src/classes_def_objects.xml: > @@ -252,8 +252,8
> @@ > <version ClassVersion="11" checksum="1239840459"/> > </class> > > -
> <class name="pat::PackedCandidate" ClassVersion="16"> > - <version
> ClassVersion="17" checksum="1633877368"/> > + <class
> name="pat::PackedCandidate" ClassVersion="17"> > + <version
> ClassVersion="17" checksum="1257500115"/> well, does it matter? I mean if
> no release exists with "17" why should we care about increasing the version
> number? — Reply to this email directly or view it on GitHub.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/8122/files#r26450034>.
>

jshlee replied:
@davidlange6 and @arizzi I'll test and update this as required.

jshlee replied:
Hi David, I've tried rebuilding and "scram b updateclassversion" and didn't see any changes. How can I get the checksum for 18?

arizzi replied:
mhh then you did already update in the last rebase the checksum including
the changed that lead to current "17" ?
if so just increase the ClassVersion to 18 keeping 1257500115 as checksum

On Sun, Mar 15, 2015 at 7:39 PM, Jason Lee <notifications@github.com> wrote:

> In DataFormats/PatCandidates/src/classes_def_objects.xml
> <https://github.com/cms-sw/cmssw/pull/8122#discussion_r26450577>:
>
> > @@ -252,8 +252,8 @@
> >     <version ClassVersion="11" checksum="1239840459"/>
> >    </class>
> >
> > -  <class name="pat::PackedCandidate" ClassVersion="16">
> > -   <version ClassVersion="17" checksum="1633877368"/>
> > +  <class name="pat::PackedCandidate" ClassVersion="17">
> > +    <version ClassVersion="17" checksum="1257500115"/>
>
> Hi David, I've tried rebuilding and "scram b updateclassversion" and
> didn't see any changes. How can I get the checksum for 18?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/8122/files#r26450577>.
>

davidlange6 commented on PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py:
```diff
@@ -174,6 +174,57 @@ def miniAOD_customizeCommon(process):
+    process.slimmedMETsPuppi.rawUncertainties   = cms.InputTag("patPFMet%s")
+    process.slimmedMETsPuppi.type1Uncertainties = cms.InputTag("patPFMetT1%s")
+    process.slimmedMETsPuppi.type1p2Uncertainties = cms.InputTag("patPFMetT1T2%s")
 
```
Could most of this go into proper cfis and cffs instead of something miniAOD specific?

arizzi replied:
I guess only this is not MINIAOD specific
    ## puppi met
    process.load('RecoMET.METProducers.PFMET_cfi')
    process.pfMetPuppi = process.pfMet.clone()
    process.pfMetPuppi.src = cms.InputTag("puppi")
    process.pfMetPuppi.alias = cms.string('pfMetPuppi')

On the other hand also the miniaod part could go in some cfi/cff ... the toplevel miniaod.py should be used only for direct access to process instance.
I think we should merge this, then me or giovanni can do a central cleanup of the configs.

davidlange6 replied:
sounds good - minimizing what is miniAODTools is best

cmsbuild commented:
+1
Tested at: ad33789565d002dd3328a2a67c065cd4c894f831
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8122/6668/summary.html

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_5_X IBs unless changes (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @nclopezo, @ktf, @smuzaffar

jshlee committed:
Update classes_def_objects.xml

updated as recommended by @davidlange6

davidlange6 commented:
+1

cmsbuild commented:
Pull request #8122 was updated. @nclopezo, @cvuosalo, @monttj, @cmsbuild, @slava77, @vadler can you please check and sign again.

cvuosalo commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cvuosalo commented:
+1

For #8122 2b4c6551da17ca04a25e1357bd5b829170a8f488

A very small change to an XML file was made after this PR was approved by Reco. The change is satisfactory and does not involve reco files.

cmsbuild commented:
+1
Tested at: 2b4c6551da17ca04a25e1357bd5b829170a8f488
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8122/3522/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8122/3522/summary.html
